### ISSUE TITLE
 cover position not in real time

### ISSUE THREAD

**Issue by superman110** [AlexxIT/SonoffLAN#517]
The cover position can't be updated in real time. It can't be correct until ha service is restarted

**Comment by albarlow** [comment]
I have similar problem.  I can open, close and stop the cover but Home Assistant doesn't know what the cover postion is and it doesn't know the cover has stopped moving when the device auto stops when it hits the limit.

**Comment by AlexxIT** [comment]
Enable debug mode and show logs

**Comment by albarlow** [comment]
> Enable debug mode and show logs

I have rebooted to get clean logs, I set a few different positions close, open, 25%, 75%, 50% and then stop (all commands sent from HA).

It doesn't know the cover position or when the motor stops.  Looking at the debug I can't see any data coming in that looks like these.

I can see that `'switch' : 'on' ` and `'switch' : 'off'` is the command to fully open or fully close and `'setclose' :` sets the percentage open and I assume that those are my requests going out(?)  There also appears to be `setclose` values coming in but this doesn't seem to make it to the front end.

If there is anything further I can provide to clarify, or can try, just let me know.

Thank You

`
2021-07-28 08:32:17  DEBUG    main          SysInfo: {'version': '2021.7.4', 'dev': False, 'hassio': True, 'virtualenv': False, 'python_version': '3.9.5', 'docker': True, 'arch': 'x86_64', 'os_name': 'Linux', 'os_version': '5.4.0-77-generic', 'supervisor': '2021.06.8', 'host_os': 'Ubuntu 20.04.2 LTS', 'docker_version': '19.03.13', 'chassis': 'laptop'}
2021-07-28 08:32:17  DEBUG    main          AUTO mode start
2021-07-28 08:32:23  DEBUG    sonoff_cloud  3 devices loaded from the Cloud Server
2021-07-28 08:32:23  DEBUG    main          10008789be == Init   | {'uiid': 11, 'extra': {'manufacturer': 'YGHOCS', 'model': 'YSEL18-1', 'sw_version': 'PSF-BTA-GL v3.4.0'}, 'params': {'rssi': -53, 'staMac': '...', 'fwVersion': '3.4.0', 'setclose': 0, 'switch': 'pause', 'version': 8, 'sledOnline': 'on', 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...', 'cloud': 'online'}}
2021-07-28 08:32:23  DEBUG    main          1000a4960b == Init   | {'uiid': 1, 'extra': {'manufacturer': 'SONOFF', 'model': 'MINI', 'sw_version': 'PSF-BD1-GL v3.6.0'}, 'params': {'version': 8, 'sledOnline': 'on', 'switch': 'on', 'fwVersion': '3.6.0', 'rssi': -52, 'staMac': '...', 'startup': 'off', 'init': 1, 'pulse': 'off', 'pulseWidth': 500, 'swMode': 2, 'swCtrlReverse': 'off', 'cloud': 'offline'}}
2021-07-28 08:32:23  DEBUG    main          1001116477 == Init   | {'uiid': 11, 'extra': {'manufacturer': 'YGHOCS', 'model': 'YSEL18-1', 'sw_version': 'PSF-BTA-GL v3.4.0'}, 'params': {'version': 8, 'sledOnline': 'on', 'ssid': '...', 'bssid': '...', 'rssi': -48, 'fwVersion': '3.4.0', 'staMac': '...', 'setclose': 50, 'switch': 'pause', 'cloud': 'online'}}
2021-07-28 08:32:24  DEBUG    sonoff_cloud  Cloud init: {'error': 0, 'apikey': '...', 'config': {'hb': 1, 'hbInterval': 145}, 'sequence': '1627457544136'}
2021-07-28 08:34:41  DEBUG    sonoff_cloud  1001116477 => Cloud4 | {'switch': 'off'} | 1627457681648
2021-07-28 08:34:42  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'config': {'switch': 'off'}, 'sequence': '1627457681648'}
2021-07-28 08:34:42  DEBUG    sonoff_cloud  1001116477 => Cloud5 | Force update sequence: 1627457682172
2021-07-28 08:34:42  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'sequence': '1627457682172', 'params': {'bindInfos': {}, 'version': 8, 'sledOnline': 'on', 'ssid': '...', 'bssid': '...', 'rssi': -48, 'fwVersion': '3.4.0', 'staMac': '...', 'setclose': 50, 'switch': 'off'}}
2021-07-28 08:34:47  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'action': 'update', 'deviceid': '1001116477', 'apikey': '...', 'userAgent': 'device', 'params': {'setclose': 50}, 'from': 'device'}
2021-07-28 08:34:52  DEBUG    sonoff_cloud  1001116477 => Cloud4 | {'switch': 'on'} | 1627457692295
2021-07-28 08:34:52  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'config': {'switch': 'on'}, 'sequence': '1627457692295'}
2021-07-28 08:34:52  DEBUG    sonoff_cloud  1001116477 => Cloud5 | Force update sequence: 1627457692853
2021-07-28 08:34:52  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'sequence': '1627457692853', 'params': {'bindInfos': {}, 'version': 8, 'sledOnline': 'on', 'ssid': '...', 'bssid': '...', 'rssi': -48, 'fwVersion': '3.4.0', 'staMac': '...', 'setclose': 50, 'switch': 'on'}}
2021-07-28 08:35:01  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'action': 'update', 'deviceid': '1001116477', 'apikey': '...', 'userAgent': 'device', 'params': {'setclose': 0}, 'from': 'device'}
2021-07-28 08:35:04  DEBUG    sonoff_cloud  1001116477 => Cloud4 | {'setclose': 74} | 1627457704217
2021-07-28 08:35:04  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'config': {'setclose': 74}, 'sequence': '1627457704217'}
2021-07-28 08:35:04  DEBUG    sonoff_cloud  1001116477 => Cloud5 | Force update sequence: 1627457704553
2021-07-28 08:35:04  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'sequence': '1627457704553', 'params': {'bindInfos': {}, 'version': 8, 'sledOnline': 'on', 'ssid': '...', 'bssid': '...', 'rssi': -48, 'fwVersion': '3.4.0', 'staMac': '...', 'setclose': 74, 'switch': 'on'}}
2021-07-28 08:35:09  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'action': 'update', 'deviceid': '1001116477', 'apikey': '...', 'userAgent': 'device', 'params': {'setclose': 74}, 'from': 'device'}
2021-07-28 08:35:11  DEBUG    sonoff_cloud  1001116477 => Cloud4 | {'setclose': 26} | 1627457711879
2021-07-28 08:35:12  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'config': {'setclose': 26}, 'sequence': '1627457711879'}
2021-07-28 08:35:12  DEBUG    sonoff_cloud  1001116477 => Cloud5 | Force update sequence: 1627457712153
2021-07-28 08:35:12  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'sequence': '1627457712153', 'params': {'bindInfos': {}, 'version': 8, 'sledOnline': 'on', 'ssid': '...', 'bssid': '...', 'rssi': -48, 'fwVersion': '3.4.0', 'staMac': '...', 'setclose': 26, 'switch': 'on'}}
2021-07-28 08:35:16  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'action': 'update', 'deviceid': '1001116477', 'apikey': '...', 'userAgent': 'device', 'params': {'setclose': 26}, 'from': 'device'}
2021-07-28 08:35:22  DEBUG    sonoff_cloud  1001116477 => Cloud4 | {'setclose': 50} | 1627457722389
2021-07-28 08:35:22  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'config': {'setclose': 50}, 'sequence': '1627457722389'}
2021-07-28 08:35:22  DEBUG    sonoff_cloud  1001116477 => Cloud5 | Force update sequence: 1627457722763
2021-07-28 08:35:22  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'sequence': '1627457722763', 'params': {'bindInfos': {}, 'version': 8, 'sledOnline': 'on', 'ssid': '...', 'bssid': '...', 'rssi': -48, 'fwVersion': '3.4.0', 'staMac': '...', 'setclose': 50, 'switch': 'on'}}
2021-07-28 08:35:25  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'action': 'update', 'deviceid': '1001116477', 'apikey': '...', 'userAgent': 'device', 'params': {'setclose': 50}, 'from': 'device'}
2021-07-28 08:35:27  DEBUG    sonoff_cloud  1001116477 => Cloud4 | {'switch': 'pause'} | 1627457727798
2021-07-28 08:35:28  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'config': {'switch': 'pause'}, 'sequence': '1627457727798'}
2021-07-28 08:35:28  DEBUG    sonoff_cloud  1001116477 => Cloud5 | Force update sequence: 1627457728265
2021-07-28 08:35:28  DEBUG    sonoff_cloud  1001116477 <= Cloud3 | {'error': 0, 'deviceid': '1001116477', 'apikey': '...', 'sequence': '1627457728265', 'params': {'bindInfos': {}, 'version': 8, 'sledOnline': 'on', 'ssid': '...', 'bssid': '...', 'rssi': -48, 'fwVersion': '3.4.0', 'staMac': '...', 'setclose': 50, 'switch': 'pause'}}
`
